http2/pom.xml

```diff
@@ -1,140 +1,126 @@
-			<version>${netty.version}</version>
-		</dependency>
-
-		<dependency>
-			<groupId>org.eclipse.jetty.http2</groupId>
-			<artifactId>http2-client</artifactId>
-			<version>${netty.version}</version>
-		</dependency>
-
-		<dependency>
-			<groupId>org.eclipse.jetty.http2</groupId>
-			<artifactId>http2-common</artifactId>
-			<version>${netty.version}</version>
-		</dependency>
-
-		<dependency>
-			<groupId>org.eclipse.jetty</groupId>
-			<artifactId>jetty-http</artifactId>
-			<version>${netty.version}</version>
-		</dependency>
-
-		<dependency>
-			<groupId>org.eclipse.jetty</groupId>
-			<artifactId>jetty-io</artifactId>
-			<version>${netty.version}</version>
-		</dependency>
-
-		<dependency>
-			<groupId>org.eclipse.jetty.http2</groupId>
-			<artifactId>http2-hpack</artifactId>
-			<version>${netty.version}</version>
-		</dependency>
-
-		<dependency>
-			<groupId>org.eclipse.jetty.osgi</groupId>
-			<artifactId>jetty-osgi-alpn</artifactId>
-			<version>${netty.version}</version>
-		</dependency>
-
-		<dependency>
-			<groupId>org.eclipse.jetty</groupId>
-			<artifactId>jetty-alpn-client</artifactId>
-			<version>${netty.version}</version>
-		</dependency>
-		
-		<dependency>
-			<groupId>org.mockito</groupId>
-			<artifactId>mockito-core</artifactId>
-			<version>2.8.47</version>
-			<scope>test</scope>
-		</dependency>
-		<dependency>
-			<groupId>junit</groupId>
-			<artifactId>junit</artifactId>
-			<version>4.12</version>
-			<scope>test</scope>
-		</dependency>
-		
-		<dependency>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.blazemeter</groupId>
+    <artifactId>jmeter-bzm-http2</artifactId>
+    <packaging>jar</packaging>
+    <version>1.1</version>
+    <name>HTTP/2 Sampler</name>
+    <description>HTTP/2 protocol sampler</description>
+    <licenses>
+        <license>
+            <name>The Apache Software License, Version 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+    <url>http://jmeter-plugins.org/</url>
+    <scm>
+        <url>https://github.com/blazemeter/jmeter-bzm-plugins</url>
+        <connection>https://github.com/blazemeter/jmeter-bzm-plugins.git</connection>
+        <developerConnection>git@github.com:blazemeter/jmeter-bzm-plugins.git</developerConnection>
+    </scm>
+    <developers>
+        <developer>
+            <id>team</id>
+            <name>jmeter-plugins.org</name>
+            <email>jmeter-plugins@googlegroups.com</email>
+        </developer>
+    </developers>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <version.org.apache.jmeter.all>3.2</version.org.apache.jmeter.all>
+        <netty.version>9.4.0.v20161208</netty.version>
```

Comment:
Some users have problems with this pluin and of their solution is update jetty dependencies. What do you think about it?
https://groups.google.com/d/msg/jmeter-plugins/ux9WvkwWb6g/-jDT3U8RAwAJ

Reply:
The reported NullPointerException is fixed by the code in these commits (no need to remove the cookie manager). The timeout exception is a different matter, I don't know exactly what are they trying to verify, but would be nice to have a jmx from them to be able to identify the actual issue. NBC was also getting TimeoutException which was caused by including HTTP headers starting with colons (ej: :method: POST) which causes (we didn't dig much on the actual cause of this since having headers with colons would generate empty named headers which don't look right) the jetty plugin to timeout. We also tested with [curl with http2 support](https://simonecarletti.com/blog/2016/01/http2-curl-macosx/) and got a `Failed sending HTTP POST request` error. Just for the record jmeter HTTP plugin doesn't fail if headers starting with colon are included.

I will in any case update jetty dependency which is most of the times a good thing to do.